l1-contracts/contracts/common/libraries/openzeppelin/IncrementalMerkle.sol

```diff
@@ -122,24 +95,31 @@ library DynamicIncrementalMerkle {
             }
```

Comment:
You do `log(n)` storage writes in the worst case on, but have `O(1)` access to the merkle root. Instead you can do 1 storage write and have access to the storage root of `log(n)` storage reads. Does your implementation intend to take this trade off?